flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  description = "My haskell library";
+
+  nixConfig = {
+    extra-substituters = [
+      "https://haskell-language-server.cachix.org"
+    ];
+    extra-trusted-public-keys = [
+      "haskell-language-server.cachix.org-1:juFfHrwkOxqIOZShtC4YC1uT1bBcq2RSvC7OMKx0Nz8="
+    ];
+  };
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    hls.url = "github:haskell/haskell-language-server/854d2c5a1aa3095c93e3f38ef3731be89b4ab09a";
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, hls, ... }:
+    flake-utils.lib.eachSystem ["x86_64-linux"] (system: let
```

Comment:
In this configuration `cabal2nix` would support at least all platforms nixpkgs GHC can run on.